sstables/version.hh

```diff
@@ -37,6 +37,11 @@ inline std::array<sstable_version_types, 4> all_sstable_versions = {
     sstable_version_types::md,
 };
 
+inline std::array<sstable_version_types, 2> writable_sstable_versions = {
+    sstable_version_types::mc,
+    sstable_version_types::md,
+};
```

Comment:
Nitpick: with this approach, every time we add a new sstable type we need to rememer to add it to both lists. It's not a huge problem of course, but maybe it will be easier if the second list to be a set (not array) of "read_only_sstable_versions" (which will list only ka and la), so if we want a list of all writable versions we can iterate over all_sstable_versions and exclude versions in read_only_sstable_versions.

Reply:
You can have an oldest_writeable_sstable_format variable, and then it need only change when we remove write support from older format.

Reply:
Since this header is not test only, maybe we could leave writable_sstables_versions as an array and just add a static assert that makes sure that writable_sstable_versions contains everything from all_sstable_versions that's above oldest_writable_sstable_format?